packages/eslint-plugin/src/rules/no-shadow.ts

```diff
@@ -595,12 +619,26 @@ export default util.createRule<Options, MessageIds>({
           ) &&
           !(options.hoist !== 'all' && isInTdz(variable, shadowed))
         ) {
+          const location = getDeclaredLocation(shadowed);
+          const report = location.global
+            ? ({
+                messageId: 'noShadowGlobal',
+                data: {
+                  name: variable.name,
+                },
+              } as const)
+            : ({
+                messageId: 'noShadow',
+                data: {
+                  name: variable.name,
+                  shadowedLine: location.line,
+                  shadowedColumn: location.column,
+                },
+              } as const);
+
           context.report({
             node: variable.identifiers[0],
-            messageId: 'noShadow',
-            data: {
-              name: variable.name,
-            },
+            ...report,
```

Comment:
Nit: the ternary could be inlined in here. Then there would be no need for the `as const`s.

```ts
context.report({
    node: variable.identifiers[0],
        ...(location.global
            ? {
                messageId: 'noShadowGlobal',
            // ...
```